### PR TITLE
AMX ID System: allow cookie storage

### PR DESCRIPTION
**Pull request by nickjacob**
update documentation

## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [X] Other

## Description of change

Changes to AMX ID System, removing requirement for HTML5/storage configuration (e.g. allowing cookie storage)

### CODE REVIEW

File: modules/amxIdSystem.md [hunk 1]
```diff
@@ -29,15 +29,15 @@ pbjs.setConfig({
...
-| --------------------- | -------- | ------------ | -------------------------------------------------------------------------------- | --------- |
-| name                  | Required | String       | Where the ID will be stored                                                      | `"amxId"` |
-| type                  | Required | String       | This must be `"html5"`                                                           | `"html5"` |
-| expires               | Required | Number <= 30 | number of days until the stored ID expires. **Must be less than or equal to 30** | `14`      |
+| Param under `storage` | Type         | Description                                                                      | Example   |
+| --------------------- |  ------------ | -------------------------------------------------------------------------------- | --------- |
+| name                  |  String       | Where the ID will be stored                                                      | `"amxId"` |
+| type                  |  String       | For best performnace, this should be `"html5"`                                                           | `"html5"` |
```
Comment (patmmccann) [inline modules/amxIdSystem.md:42]: typo

File: modules/amxIdSystem.js [hunk 2]
```diff
@@ -97,6 +107,28 @@ export const amxIdSubmodule = {
       ? { [ID_KEY]: value }
       : undefined,
 
+  /**
+   * Similar to sharedIdSystem domainOverride
+   * if on a subdomain, try to set the cookie on the root domain
+   */
+  domainOverride() {
```
[the remark this answers is not in the document]
Comment (nickjacob) [reply modules/amxIdSystem.js:114]: Good idea -- I can extract the function from sharedIdSystem into utils or somewhere in src. Goal here is to try to set the cookie on the topmost possible domain/root domain

[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/amxIdSystem.js:114]: with https://github.com/prebid/Prebid.js/pull/8527 and #8599 we have libraries you can import from

**Comment by nickjacob** [comment]
refactored into a library -- "domainOverrideToRootDomain" using code from sharedId

**Review by smenzer — APPROVED** [review]
LGTM.

@patmmccann do we need a 2nd reviewer since there is a library added? or are libraries separate from "core" that requires 2?

**Review by patmmccann — APPROVED** [review]

**Comment by patmmccann** [comment]
Changes lgtm

**Comment by patmmccann** [comment]
We see to have had a conflict arise, perhaps with the work @dgirardi has been doing on storage

**Comment by nickjacob** [comment]
@patmmccann rebased & updated my call to `getStorageManager` to work with new storage code

**Pull request merged**